docs/source/guide/core-concepts.myst

```diff
@@ -0,0 +1,76 @@
+For hardware, the results can only be binary values for each evaluation of the program.
+For simulators, more information can be made available like the resulting density matrices or state vectors, which will be valid values of the {class}`.QuantumResult` type.
+
+## Frontends
+
+Mitiq can accept quantum circuit representations in a variety of formats from different tools or platforms.
+These input formats are referred to here as _frontends_.
+The frameworks currently supported are:
+
+```{code-cell} ipython3
+import mitiq
+
+mitiq.SUPPORTED_PROGRAM_TYPES.keys()
+```
+
+with Cirq being used for the internal implementations of the mitigation methods.
+There is also frontend support for any circuit description that complies with the OpenQASM 2.0 standard.
+
+With any of these frontends (with the exception of Cirq), the circuit representation will be converted internally to a Cirq circuit object with the relevant conversion methods in {py:mod}`mitiq.interface` for each frontend.
+For example, you can use {func}`mitiq.interface.mitiq_braket.conversions.from_braket()` to convert a Braket circuit to a Cirq circuit.
+
+Examples for using each of the frameworks to represent the input to a mitigation method are linked below.
+
+- {ref}`Cirq <examples/ibmq-backends/cirq-frontend>`
+- {ref}`Qiskit <examples/ibmq-backends/setup-defining-a-circuit>`
+- {ref}`PyQuil <examples/pyquil_demo>`
+- {ref}`Braket <examples/braket_mirror_circuit/define-the-circuit>`
+
+## Backends
+
+A _backend_ is any simulator or hardware device that can be used to execute programs that are valid `mitiq.QPROGRAM` objects.
+These backends are usually installed separately as needed by the user.
+Backends are used by Mitiq in the {class}`.Executor` class to run the mitigated `mitiq.QPROGRAM` objects.
+
+## Executing programs
+
+Once you have selected a frontend and backend combination that are compatible and the mitigation method, the next step is to run the mitigated program.
+The {class}`.Executor` class is used to execute the mitigated program, and is constructed by providing a function with the following signature: `mitiq.QPROGRAM` -> {class}`.QuantumResult`.
+For more information on executors, see the {doc}`Executors <executor>` section of this guide.
+
+### Batch execution of programs
+
+When you have a number of circuits that you want to execute with an error mitigation method, you can use the {class}`.Executor` class to execute them all in one go.
+This is done by providing a list of `mitiq.QPROGRAM` objects to the {class}`.Executor` constructor.
+The {class}`.Executor` class will then execute all of the programs in parallel, and return a list of {class}`.QuantumResult` objects.
+To perform batched execution, you can use the same mitiq.Executor class constructor and pass a function with the following signature: `T[mitiq.QPROGRAM] -> T[{class}".QuantumResult"]` where T is `Sequence`, `List`, `Tuple`, or `Iterable`.
```

Comment:
```suggestion
You can also use the {class}`.Executor` class to execute a batch of quantum programs all in one go.
To perform batched execution, you can use the same mitiq.Executor class constructor and pass a function with the following signature: `T[mitiq.QPROGRAM] -> T[{class}".QuantumResult"]` where T is `Sequence`, `List`, `Tuple`, or `Iterable`.
```
I would suggest this simplified text. 
Note that what can be batched is only the execution of noisy unmitigated circuits. The final error mitigation is still defined for a single circuit.